### PR TITLE
Hologram module overhaul

### PR DESCRIPTION
**Pull request by LeeGodSRC**
Hologram module is now cross-platform capable (depend on mc-platform rather than bukkit-platform)
Added Pos class and desperated CustomLocation
Added MCPlayerMoveEvent
Added MCPlayerTeleportEvent
Added MCWorldUnloadEvent
Added PacketListener Obj scanning
Added more functions to MCServer, MCWorld, MCPlayer, MCEntity

## TODO:
- [x] Documentation
- [x] Testing
- [x] Code style fixes

### CODE REVIEW

**Comment by terminalsin** [comment]
Awaiting codacy fix

File: io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.github.retrooper.packetevents.event.ProtocolPacketEvent;
+import com.github.retrooper.packetevents.manager.server.ServerVersion;
+import com.github.retrooper.packetevents.protocol.ConnectionState;
+import com.github.retrooper.packetevents.protocol.packettype.PacketTypeCommon;
+import io.fairyproject.event.Cancellable;
+import io.fairyproject.mc.MCPlayer;
+import io.fairyproject.mc.event.trait.MCPlayerEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class MCPlayerProtocolPacketEvent implements MCPlayerEvent, Cancellable {
```
Comment (Whizyyy) [inline io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java:12]: Why keyword "get" is never used?

File: io.fairyproject.platforms/mc-platform/src/main/java/io/fairyproject/mc/util/Pos.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+/**
+ * @since 0.6.2b1-SNAPSHOT
+ */
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@JsonSerialize(using = Pos.Serializer.class)
+@JsonDeserialize(using = Pos.Deserializer.class)
+public class Pos implements Cloneable {
```
Comment (Whizyyy) [inline io.fairyproject.platforms/mc-platform/src/main/java/io/fairyproject/mc/util/Pos.java]: Why not call it Position to be more explicit?

File: io.fairyproject.modules/platform.mc/module.hologram/src/main/java/io/fairyproject/mc/hologram/HologramImpl.java [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+        this.lines.remove(index);
+    }
+
+    @Override
+    public void clear() {
+        this.lines.clear();
+    }
+
+    @Override
+    public Hologram spawn() {
```
Comment (Whizyyy) [inline io.fairyproject.modules/platform.mc/module.hologram/src/main/java/io/fairyproject/mc/hologram/HologramImpl.java:152]: If im right, you never verify that the "pos" is not null which is gonna throw NPE in the future

File: io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java (hunk 1 above)
Comment (LeeGodSRC) [reply io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java:12]: What do you mean by keyword "get?"

File: io.fairyproject.platforms/mc-platform/src/main/java/io/fairyproject/mc/util/Pos.java (hunk 2 above)
Comment (LeeGodSRC) [reply io.fairyproject.platforms/mc-platform/src/main/java/io/fairyproject/mc/util/Pos.java]: for simply, bug i guess you are right.

File: io.fairyproject.modules/platform.mc/module.hologram/src/main/java/io/fairyproject/mc/hologram/HologramImpl.java (hunk 3 above)
Comment (LeeGodSRC) [reply io.fairyproject.modules/platform.mc/module.hologram/src/main/java/io/fairyproject/mc/hologram/HologramImpl.java:152]: fixed

File: io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java (hunk 1 above)
Comment (Whizyyy) [reply io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java:12]: Methods should contain a verb, for a lot of your getters in this PR the method miss "get" keyword, example: pos() or player() are just position & player getters but this do not really follow Java convention tbh

Comment (LeeGodSRC) [reply io.fairyproject.modules/platform.mc/module.protocol/src/main/java/io/fairyproject/mc/protocol/event/MCPlayerProtocolPacketEvent.java:12]: For some reason I was fall in love to the idea of everything "fluent" including getters and setters.
But I guess you are right, stop fluent getter.

**Review by Whizyyy — APPROVED** [review]

**Pull request merged**